### PR TITLE
Windows fix

### PR DESCRIPTION
**Pull request by alsaiduq-lab**
lol windows fix

- added .dll cause windows wants it that way
- updated .gitignore for windows

was trying to use this on windows and it didn't work so here's the fix 😅

### CODE REVIEW

**Comment by vyfor** [comment]
> was trying to use this on windows and it didn't work

Were there any errors either in `:messages` or the plugin manager?

**Comment by vyfor** [comment]
That likely means that the build script either ran with an error or did not run at all. You could try manually running the `build.bat` file.

**Comment by alsaiduq-lab** [comment]
I ran the build.bat manually and found the issue - without the .dll extension, Windows creates cord.exe instead of cord.dll:

```powershell
❯ ./build.bat
C:\Users\Mon_A\AppData\Local\nvim-data\lazy\cord.nvim>rustc --crate-type=cdylib -C opt-level=3 -C panic=abort -C lto=fat -o cord src/lib.rs

❯ ls cord*
-a---     2024-12-03  05:42          1.98 MB     cord.exe

**Comment by vyfor** [comment]
What Rust version are you using? `rustc --version -v`

**Comment by alsaiduq-lab** [comment]
might be because im using the GNU toolchain

❯ rustc --version -v
rustc 1.82.0 (f6e511eec 2024-10-15)
binary: rustc
commit-hash: f6e511eec7342f59a25f7c0534f1dbea00d01b14
commit-date: 2024-10-15
host: x86_64-pc-windows-gnu
release: 1.82.0
LLVM version: 19.1.1

**Comment by vyfor** [comment]
I can confirm that the GNU toolchain does append `.exe` even when the extension isn’t specified. Could you open an issue for this, or would you prefer me to do it?

**Comment by vyfor** [comment]
Also, passing `-o cord.` seems to get rid of any extensions.

**Comment by alsaiduq-lab** [comment]
I have confirmed that `-o cord.` worked perfectly (I cannot believe this worked lol).

**Comment by alsaiduq-lab** [comment]
You can open if you'd like, but this fix is more easier.

**Review by vyfor — CHANGES_REQUESTED** [review]
It is necessary for it to be compiled without any extensions.
I'd be happy to merge the PR if you can apply the requested changes 🙂. Also, could you override your commits and update their messages to align with the [spec](https://github.com/vyfor/cord.nvim/blob/master/.github/CONTRIBUTING.md#git-commit-messages)?
P.S. `.gitignore` would belong in `chore` and `build.bat` in `build` types

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,5 @@
 /cord
 /cord.*
 .direnv/
+*.a
+*.dll.a
```
Comment (vyfor) [inline .gitignore:7]:
```suggestion

```

`*.a` already covers it.

File: build.bat [hunk 2]
```diff
@@ -1 +1 @@
-rustc --crate-type=cdylib -C opt-level=3 -C panic=abort -C lto=fat -o cord src/lib.rs
\ No newline at end of file
+rustc --crate-type=cdylib -C opt-level=3 -C panic=abort -C lto=fat -o cord.dll src/lib.rs
```
Comment (vyfor) [inline build.bat:1]:
```suggestion
rustc --crate-type=cdylib -C opt-level=3 -C panic=abort -C lto=fat -o cord. src/lib.rs
```

**Review by vyfor — COMMENTED** [review]
GH is being silly.

**Comment by alsaiduq-lab** [comment]
Sorry, I should've read more. Is this good?

**Comment by vyfor** [comment]
That works, thank you!

**Pull request merged**